tools/test/examples/README.md

```diff
@@ -0,0 +1,167 @@
+* **compile** - enables compiling.
+* **export** - enables exporting.
+* **test** - enables testing.
+
+### Values
+
+`[ ]` means all possible alternatives.
+
+## Typical use
+
+In the Mbed OS CI, we follow the below steps to compile and test Mbed OS examples.
+
+1. Clone `mbed-os` repository to the current folder:
+
+   ```
+   git clone https://github.com/ARMmbed/mbed-os.git
+   ```
+
+1. Clone the examples repo to the current folder. Users can pass an `-e` option to the script to filter out the rest of the examples, so the scripts only run on one particular example:
+
+   ```
+   python mbed-os/tools/test/examples/examples.py clone
+   ```
+
+1. Create a symbolic link to `mbed-os` for every example. This step lets all the examples share a single `mbed-os` folder, rather than checking out the `mbed-os` folder many times. We highly recommend you pass an absolute path as the argument: 
+
+   ```
+   python mbed-os/tools/test/examples/examples.py symlink $PWD/mbed-os
+   ```
+
+1. Deploy other dependency libraries:
+
+   ```
+   python mbed-os/tools/test/examples/examples.py deploy
+   ```
+
+1. Compile the test for the examples on a specific target:
+
+   ```
+   python mbed-os/tools/test/examples/examples.py compile -m <target>
+   ```
+
+After the compile test finished, the scripts print the result table:
+
+```
+Passed example compilation:
++---------------------------------+--------+-----------+----------+--------------+
+| EXAMPLE NAME                    | TARGET | TOOLCHAIN | TEST GEN | BUILD RESULT |
++---------------------------------+--------+-----------+----------+--------------+
+| mbed-os-example-kvstore         |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-tls-socket      |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-blockdevice     |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-wifi            |  K64F  |  GCC_ARM  | TEST_OFF |    PASSED    |
+| mbed-os-example-error-handling  |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-sd-driver       |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-crash-reporting |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-filesystem      |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-blinky          |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-bootloader      |  K64F  |  GCC_ARM  | TEST_OFF |    PASSED    |
+| mbed-os-example-cpu-stats       |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-sys-info        |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
+| mbed-os-example-attestation     |  K64F  |  GCC_ARM  | TEST_ON  |    PASSED    |
++---------------------------------+--------+-----------+----------+--------------+
+Number of failures = 0
+```
+
+After the compilation stage, a `test_spec.json` file is generated. Later, Greentea tests will consume this file. They test the compiled example on hardware platform. 
```

Comment:
Can this `test_spec.json` also include some specific test configuration settings, for example if we want to test bare metal?

Reply:
this `test_spec.json` is generated after the compilation is done. it contains the if where is the binary file, where is the compare_log.  in terms of bare-metal profile it need to be passed in to the compilation as `mbed_apps.json`. If we have the baremetal example, it can test the baremetal example without any issues